write DOM

glact1973 opened the pull request:
HTMLを出力するDOM要素を記載しました。

glact1973 committed:
write DOM

HTMLを出力するDOM要素を記載しました。

Mitsuru53 commented:
こちら、
エラーが出て動作確認ができないため、
修正のよろしくお願いいたします！
<img width="1670" alt="2018-12-01 14 18 35" src="https://user-images.githubusercontent.com/23156390/49324631-e4e4a000-f574-11e8-97f0-47f1712283ae.png">

glact1973 committed:
DOMの作成

Mitsuru53 commented:
エラーはでなくなりました！
ただ画像が表示されません
また、「次へ」ボタンを押しても次の画像やテキストに変わりません
ご確認よろしくお願いします
<img width="1343" alt="2018-12-08 14 12 54" src="https://user-images.githubusercontent.com/23156390/49682193-60090180-faf3-11e8-97f3-0e88f5c6990e.png">

glact1973 committed:
Viewの変更（1つ目のデータまで表示できるようになったが、「次へ」が動作しない）

memberlistから写真、名前、コメントを出せるようにViewを変更しました。
ただし、「次へ」のボタンを押すと、addEventLisnerを使って、次のデータを表示するよう、恐らく'BrowseMembers.js'の'incrementCurrentMember()'を読み込むのだと思います。
が、うまくいきません。
まず。addEventListenerは、DOMを書くところで書くべきかと考え、BrowseMembersViewの'const btnNextMember'の一部で書いています。しかしながら、そうすると、incrementCurrentMember()を読み出せません（class BrowseMembersView はBrowseMembers.jsにimportされるものであり、逆のimportをすると違和感があるし、試しても動かなかった）。

かといって、BrowseMembers.jsでaddEventListenerをしようと↓のように書いてみたが、"btn.addEventListener is not a function"のエラーになる。（動いていないので、アップしていません）
  patchState(patch) {
    Object.assign(this.state, patch);
    this.view.render(this.mergeProps());
    // 試しに追加
    var btn = document.getElementsByClassName('btn-next-member');
    btn.addEventListener('click',() => {
      incrementCurrentMember(membersList);
    }); 
    // 追加ここまで
  }

「次へ」の処理はどのように記載すべきでしょうか・・・？

glact1973 committed:
「次へ」でリンクするように変更

・「次へ」ボタンで次の処理に移動するように設定
・改行コードをLFに変更

glact1973 committed:
余計なコメントが残っていたので削除

コード本体は先程と同じです。

glact1973 committed:
余計なコメントが入っていたため削除

glact1973 committed:
アップする場所間違えたので削除

glact1973 committed:
「次へ」の処理を追加

・「次へ」を押すと次の画像に遷移するように改良。
・改行コードをLFに変更

clonable-eden commented:
@glact1973 LGTMです！
